application/forms/RotationConfigForm.php

```diff
@@ -312,17 +315,19 @@ private function createRotation(int $priority): Generator
 
         $changedAt = (int) (new DateTime())->format("Uv");
 
-        $rotationsToMove = Rotation::on($this->db)
-            ->columns('id')
-            ->filter(Filter::equal('schedule_id', $this->scheduleId))
-            ->orderBy('priority', SORT_DESC);
-
-        foreach ($rotationsToMove as $rotation) {
-            $this->db->update(
-                'rotation',
-                ['priority' => new Expression('priority + 1'), 'changed_at' => $changedAt],
-                ['id = ?' => $rotation->id]
-            );
+        if ($priority === null) {
```

Comment:
Honestly, the need for this fix should have told you that the location of this piece of code is incorrect.

The only time the priority is null now is when a rotation is created from scratch. And the need to relocate a rotation is also not a new thing. Removing one, reordered them for example. Which is why this method is a generator: To perform preparations.

And this here is such a preparation. Just put this inside `addRotation`.